### PR TITLE
[ja] add preprocessor for width of alphabetic characters

### LINKED ISSUE #961: Words using fullwidth alphabet cannot be searched via halfwidth (normal) alphabet search strings
**Description**
In the search mode, if a word is registered in a dictionary using full-width alphabet, then searching for the word using a half-width (normal ASCII) alphabet search string will yield no results. The reverse is also true.

For example, there's an entry for `ＵＳＢ` (full-width) in JMdict (in particular, in [Jitendex](https://jitendex.org) for Yomitan), but searching in Yomitan with `USB` (half-width) will not return this result.

On the other hand, with Kenkyusha's *Shinwaei (New Japanese-English) Dictionary* (which can be obtained from *Shoui's Yomitan Dictionary Collection* via [this website](https://learnjapanese.moe/yomichan/)), searching in Yomitan with `ＵＳＢ` (full-width) does not return any results either, because there the word is registered as `USB` (half-width)

As the user usually doesn't know how a word is registered inside a particular dictionary, having Yomichan deal with this issue automatically would be great.

**Browser version**
Firefox 126.0 (64-bit)

**Yomitan version**
24.5.5.0

### PR DESCRIPTION
**Pull request by StefanVukovic99**
Fixes #961

### CODE REVIEW

File: ext/js/language/ja/japanese.js [hunk 1]
```diff
@@ -523,16 +524,38 @@ export function convertHiraganaToKatakana(text) {
...
+}
+
+/**
+ * @param {string} text
+ * @returns {string}
+ */
+export function convertFullWidthAlphanumericToNormal(text) {
+    let result = '';
+    for (const char of text) {
+        let c = /** @type {number} */ (char.codePointAt(0));
```
Comment (Kuuuube) [inline ext/js/language/ja/japanese.js:550]:
```suggestion
    let i = 0;
    const length = text.length;
    for (let i = 0; i < length; i++) {
        let c = /** @type {number} */ (text[i].codePointAt(0));
```

This type of for loop is faster. (can also do similar for the other function)

File: ext/js/language/ja/japanese.js [hunk 2]
```diff
@@ -523,16 +524,40 @@ export function convertHiraganaToKatakana(text) {
...
+    return result;
+}
+
+/**
+ * @param {string} text
+ * @returns {string}
+ */
+export function convertFullWidthAlphanumericToNormal(text) {
+    let result = '';
+    let i = 0;
```
Comment (Kuuuube) [inline ext/js/language/ja/japanese.js:549]:
```suggestion

```

oops

**Comment by StefanVukovic99** [comment]
Performance on this is OK now since we merged #973, ready for review :pray:

File: ext/js/language/ja/japanese.js [hunk 3]
```diff
@@ -523,16 +524,39 @@ export function convertHiraganaToKatakana(text) {
  * @param {string} text
  * @returns {string}
  */
-export function convertNumericToFullWidth(text) {
+export function convertAlphanumericToFullWidth(text) {
     let result = '';
```
Comment (jamesmaa) [inline ext/js/language/ja/japanese.js:528]: Would this be more performant using an array than string concatenation? Not sure if this is the performance bottleneck

**Review by jamesmaa — APPROVED** [review]

**Pull request merged**